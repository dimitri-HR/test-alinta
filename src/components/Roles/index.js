import React, { Fragment } from 'react'
import { arrayOf, string } from 'prop-types'

const Roles = ({ roles }) => (
  <Fragment>
    {roles.map(role => (
      <li key={role}>
        {role}
      </li>
    ))}
  </Fragment>
)

Roles.propTypes = {
  roles: arrayOf(string),
}

Roles.defaultProps = {
  roles: [],
}

export default Roles
