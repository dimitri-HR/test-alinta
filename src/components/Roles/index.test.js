import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Roles from '.'

describe('Roles', () => {
  it('renders without errors', () => {
    const component = renderer.create(<Roles />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders correct number of roles', () => {
    const mockRoles = ['Character1', 'Character2']

    const component = mount(
      <Roles roles={mockRoles}/>
    )

    expect(component.find('li').length).toBe(2)
  })
})
