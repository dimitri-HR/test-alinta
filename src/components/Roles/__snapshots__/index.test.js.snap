// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roles renders without errors 1`] = `null`;
